hasjob/models/filterset.py

```diff
@@ -0,0 +1,159 @@
+    db.Column('filterset_id', None, db.ForeignKey('filterset.id'), primary_key=True),
+    db.Column('jobcategory_id', None, db.ForeignKey('jobcategory.id'), primary_key=True),
+    db.Column('created_at', db.DateTime, nullable=False, default=db.func.utcnow())
+)
+
+
+filterset_tag_table = db.Table('filterset_tag_table', db.Model.metadata,
+    db.Column('filterset_id', None, db.ForeignKey('filterset.id'), primary_key=True),
+    db.Column('tag_id', None, db.ForeignKey('tag.id'), primary_key=True),
+    db.Column('created_at', db.DateTime, nullable=False, default=db.func.utcnow())
+)
+
+filterset_domain_table = db.Table('filterset_domain_table', db.Model.metadata,
+    db.Column('filterset_id', None, db.ForeignKey('filterset.id'), primary_key=True),
+    db.Column('domain_id', None, db.ForeignKey('domain.id'), primary_key=True),
+    db.Column('created_at', db.DateTime, nullable=False, default=db.func.utcnow())
+)
+
+
+class FilterSet(BaseScopedNameMixin, db.Model):
+    """
+    Store filters to display a filtered set of jobs scoped by a board on SEO friendly URLs
+
+    Eg: `https://hasjob.co/f/machine-learning-jobs-in-bangalore`
+    """
+
+    __tablename__ = 'filterset'
+
+    board_id = db.Column(None, db.ForeignKey('board.id'), nullable=False, index=True)
+    board = db.relationship(Board)
+    parent = db.synonym('board')
+
+    #: Welcome text
+    description = db.Column(db.UnicodeText, nullable=False, default=u'')
+
+    #: Associated job types
+    types = db.relationship(JobType, secondary=filterset_jobtype_table)
+    #: Associated job categories
+    categories = db.relationship(JobCategory, secondary=filterset_jobcategory_table)
+    tags = db.relationship(Tag, secondary=filterset_tag_table)
+    domains = db.relationship(Domain, secondary=filterset_domain_table)
+    location_geonameids = db.Column(postgresql.ARRAY(db.Integer(), dimensions=1), nullable=True, index=True)
+    remote_location = db.Column(db.Boolean, default=False, nullable=False, index=True)
+    pay_currency = db.Column(db.CHAR(3), nullable=True, index=True)
+    pay_cash_min = db.Column(db.Integer, nullable=True, index=True)
+    equity = db.Column(db.Boolean, nullable=False, default=False, index=True)
+    keywords = db.Column(db.Unicode(250), nullable=False, default=u'', index=True)
+
+    def __repr__(self):
+        return '<FilterSet %s "%s">' % (self.board.title, self.title)
+
+    @classmethod
+    def get(cls, board, name):
+        return cls.query.filter(cls.board == board, cls.name == name).first_or_404()
+
+    def url_for(self, action='view', _external=False, **kwargs):
+        if action == 'view':
+            subdomain = self.board.name if self.board.not_root else None
+            return url_for('filterset_view', subdomain=subdomain, name=self.name, _external=_external)
+
+    def to_filters(self, translate_geonameids=False):
+        from hasjob.views.helper import location_geodata
```

Comment:
Models should never import from views, so that function has to move elsewhere. Maybe into a `hasjob.utils`.

Reply:
Should it move to `hasjob.extapi`? Since that's the convention we've been using of late for interfacing with external services.

Reply:
Yes